src/context/auth.jsx

```diff
@@ -165,4 +187,24 @@ const ProtectedRoute = ({ children, disabledNav = false }) => {
     )
 }
 
+function validationEmail(email) {
+    const emailPattern = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+  
+    return !email || !emailPattern.test(email) ? false : true;
+  }
+  
+  function validationPassword(password) {
+    const minLength = 8;
+    const hasUppercase = /[A-Z]/.test(password);
+    const hasNumber = /\d/.test(password);
+    const hasSpecialCharacter = /[!@#$%^&*(),.?":{}|<>]/.test(password);
+  
+    return password.length < minLength ||
+      !hasUppercase ||
+      !hasNumber ||
+      !hasSpecialCharacter
+      ? false
+      : true;
+  }
```

Comment:
All of these conditions will result in a true or false, you don't need to use a ternary to translate them aslong as you use the correct condition checks